### ISSUE TITLE
Classes inheriting from `ActiveType::Object` require a DB connection at load time

### ISSUE THREAD

**Issue by remofritzsche** [makandra/active_type#85]
Hi There

We're currently developing a Gem that, as most Gems, loads its source at boot time. In this source, there are a few files that specify classes inheriting from `ActiveType::Object`. When these classes are loaded at runtime (i.e. in an initializer), it works fine. But as we're loading these classes at Gem load time, it already expects a database connection and fails as follows:

```
Bundler::GemRequireError: There was an error while trying to load the gem 'our_gem'.
Gem Load Error is: No connection pool with id primary found.
Backtrace for gem load error is:
/work/path/path/vendor/gems/gems/activerecord-5.0.1/lib/active_record/connection_handling.rb:124:in `connection_pool'
/work/path/path/vendor/gems/gems/active_type-0.7.0/lib/active_type/type_caster.rb:110:in `lookup'
/work/path/path/vendor/gems/gems/active_type-0.7.0/lib/active_type/type_caster.rb:97:in `initialize'
/work/path/path/vendor/gems/gems/active_type-0.7.0/lib/active_type/type_caster.rb:10:in `get'
/work/path/path/vendor/gems/gems/active_type-0.7.0/lib/active_type/virtual_attributes.rb:49:in `add_virtual_column'
/work/path/path/vendor/gems/gems/active_type-0.7.0/lib/active_type/virtual_attributes.rb:40:in `build'
/work/path/path/vendor/gems/gems/active_type-0.7.0/lib/active_type/virtual_attributes.rb:250:in `attribute'
/work/gems/our_gem/lib/our_gem/configuration.rb:2:in `<class:Configuration>'
[...]
```

```ruby
# /work/gems/our_gem/lib/our_gem/configuration.rb
module OurGem
  class Configuration < ActiveType::Object
  end
end
```

Is it really necessary for this to load a DB connection, and this already at time of evaluating classes inheriting from the `ActiveType::Object` base class? This makes it impossible loading library code at Gem load time and also makes it impossible to use `active_type` without a database.

Thanks a bunch for having a look at it and, in general, thanks to the development of your wonderful Gem - I use it everyday.

Cheers
Remo

**Comment by triskweline** [comment]
Is it possible to inherit from `ActiveRecord::Base` without a database connection? Since we're using that under the hood.

**Comment by kratob** [comment]
I think the inheritance is not the issue, we already monkey patch that. The Rails-5-specific coercion logic is what breaks and it really shouldn't. I'll take a look why this happens.

**Comment by kratob** [comment]
This should be fixed in 0.7.1.

**Comment by remofritzsche** [comment]
Brilliant, thanks a bunch. It works like a charm and has just been integrated.
